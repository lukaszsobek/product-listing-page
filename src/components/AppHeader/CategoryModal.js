import React, { Component } from "react";
import { connect } from "react-redux";

import { updatePage, toggleModal } from "../../actions";
import Dropdown from "./Dropdown";

class CategoryModal extends Component {
    render() {
        const { data, isOpen, updatePage, toggleModal } = this.props;
        return (
            <div className={
                    isOpen 
                    ? "modal--category modal--category--open"
                    : "modal--category"
                }>
                <div
                    className="modal--category__title"
                    onClick={toggleModal}
                    >Category</div>
                <Dropdown
                    isOpen={isOpen}
                    data={data}
                    updatePage={updatePage}
                    dropdownClass="modal--category__data"
                    />
            </div>
        );
    }
};

const mapStateToProps = state => ({
    isOpen: state.modalState.categoryModalVisible,
    data: state.dropdownData.categories
});

const mapDispatchToProps = dispatch => ({
    toggleModal: () => dispatch(toggleModal("CategoryModal")),
    updatePage: (value) => dispatch(updatePage("categories",value))
});

export { CategoryModal };
export default connect(mapStateToProps, mapDispatchToProps)(CategoryModal);