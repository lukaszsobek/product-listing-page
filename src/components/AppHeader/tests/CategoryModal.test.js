import { CategoryModal } from '../CategoryModal'

describe('CategoryModal', () => {
  it('renders CategoryModal', () => {
    const props = {
      data: [1, 2],
      isOpen: false,
      updatePage: () => null,
      toggleModal: () => null
    }
    expect(new CategoryModal(props).render()).toMatchSnapshot()
  })
})
