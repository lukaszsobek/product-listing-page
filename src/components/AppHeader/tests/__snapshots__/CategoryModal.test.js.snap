// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryModal renders CategoryModal 1`] = `
<div
  className="modal--category"
>
  <div
    className="modal--category__title"
    onClick={[Function]}
  >
    Category
  </div>
  <Dropdown
    data={
      Array [
        1,
        2,
      ]
    }
    dropdownClass="modal--category__data"
    isOpen={false}
    updatePage={[Function]}
  />
</div>
`;
